### PR TITLE
message backend

### PR DESCRIPTION
**Pull request by Morvin-Ian**
Accepting a rejection message if a soc is rejected.

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -289,12 +289,25 @@ def finalize_or_reject(request, process_id):
...
+
     check, current_state = is_finalized(item_id, item_type)
     if check and current_state != "processing":
         return Response(
             f"document already processed as `{current_state}`!", status.HTTP_200_OK
         )
-    res = json.loads(finalize_item(item_id, state, item_type))
+    
+    res = json.loads(finalize_item(item_id, state, item_type, message))
+    print(res)
```
Comment (WawinyEdwin) [inline app/views.py:310]: Do we need this??

File: app/views.py [hunk 2]
```diff
@@ -289,12 +289,25 @@ def finalize_or_reject(request, process_id):
...
     state = request.data["action"]
+
+    if state == "rejected":
+        message = request.data.get("rejectionMsg", None)
+        if not message:
+            return Response(
+                "provide a reason for rejecting the document",
+                status = status.HTTP_400_BAD_REQUEST
+            )
+    else:
```
Comment (WawinyEdwin) [inline app/views.py:300]: Instead of this block first assign the message to an empty string before hand.

File: app/views.py [hunk 3]
```diff
@@ -289,12 +289,25 @@ def finalize_or_reject(request, process_id):
     user = request.data["authorized"]
     user_type = request.data["user_type"]
     state = request.data["action"]
+
+    if state == "rejected":
+        message = request.data.get("rejectionMsg", None)
```
Comment (WawinyEdwin) [inline app/views.py:294]: change this request body key to the same naming style as other variables "rejection_message"

File: app/mongo_db_connection.py [hunk 4]
```diff
@@ -819,7 +819,7 @@ def authorize(document_id, viewers, process_id, item_type):
     return
 
 
-def finalize_item(item_id, state, item_type):
+def finalize_item(item_id, state, item_type, message):
```
Comment (WawinyEdwin) [inline app/mongo_db_connection.py:822]: Confirm all the item_types that get affected especially if we have `clone` also. Clone is a document copy. Ask Uche and Couzy for Clarity on this

**Review by WawinyEdwin — CHANGES_REQUESTED** [review]

**Review by WawinyEdwin — APPROVED** [review]

**Review by WawinyEdwin — APPROVED** [review]

File: app/views.py (hunk 1 above)
Comment (Morvin-Ian) [reply app/views.py]: Removed the print

File: app/views.py (hunk 2 above)
Comment (Morvin-Ian) [reply app/views.py]: Assigned the empty message string before the block

File: app/views.py (hunk 3 above)
Comment (Morvin-Ian) [reply app/views.py]: Done.

File: app/mongo_db_connection.py (hunk 4 above)
Comment (Morvin-Ian) [reply app/mongo_db_connection.py:822]: Done

**Pull request merged**